crates/sui-framework/packages/sui-framework/sources/transfer.move

```diff
@@ -57,6 +65,51 @@ public fun public_transfer<T: key + store>(obj: T, recipient: address) {
     transfer_impl(obj, recipient)
 }
 
+/// NOT YET SUPPORTED. The function will abort with `ENotSupported` if used on a network,
+/// e.g. mainnet, where multiparty objects other than `legacy_shared` are not yet supported.
```

Comment:
`legacy_shared` is not supported on mainnet either, only actual Shared objects are supported on mainnet

I think these are not the same thing because a ConsensusV2 object that has equivalent permissions to a Shared object could be transferred to that ownership from an existing object, whereas a Shared object must be shared in the same tx that created it

Reply:
`legacy_shared` here dispatches out to `share_object` so it won't support "upgrades" yet.
This was added to stress test the API to make sure it felt like it was in the correct place for future additions